tests/ui/test_ui.py

```diff
@@ -135,17 +139,17 @@ def test_main_window(self, mocker):
         view = View(self.controller)
 
         left.assert_called_once_with()
-        center.assert_called_once_with()
+        # NOTE: Don't check center here, as we're monkeypatching it
         right.assert_called_once_with()
 
         expected_column_calls = [
             mocker.call([
                 (View.LEFT_WIDTH, left()),
-                ('weight', 10, center()),
+                ('weight', 10, mocker.ANY),  # ANY is a center
```

Comment:
* `mocker.ANY` matches all values passed in. In some cases you can get away with `.called`, not caring about the values, and in others you care about each parameter; if you can't test or don't care about one, but want to match the others, you can use this for the `anything` parameter.
* The challenge in this test is that I added the `assert` you referred to and queried the location of in another comment, which relies on `message_view` setting `msg_list`. if we want to check `message_view` is called then we need the equivalent of a mock, but we need to set the view attribute before the `__init__` is called (to avoid the `assert` failing), which depends on the existence of the object to set - or at least that's my interpretation. With `monkeypatch` we replace the method itself, but then can't check the mock. Other ideas welcome!

Reply:
Ah, I understand it now. Thanks for the clarification and the comments.